### PR TITLE
Hotfix scrm 302 sonar

### PR DESCRIPTION
**Pull request by gymad**
## Description

Fix for SonarQube issues
(ref: SCRM-311)

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Final checklist

- [ ] My code follows the code style of this project found [here](https://suitecrm.com/wiki/index.php/Coding_Standards).
- [ ] My change requires a change to the documentation.
- [ ] I have read the [**How to Contribute**](https://suitecrm.com/wiki/index.php/Contributing_to_SuiteCRM) guidelines.

### CODE REVIEW

File: modules/Tasks/metadata/additionalDetails.php [hunk 1]
```diff
@@ -40,7 +40,7 @@
 
 require_once "include/utils/additional_details.php";
 
-function additionalDetailsTask($fields = array(), SugarBean $bean, $params = array()) {
+function additionalDetailsTask($fields = array(), SugarBean $bean, $params) {
```
Comment (mattlorimer) [inline modules/Tasks/metadata/additionalDetails.php]: wrong array() remove?

Comment (gymad) [inline modules/Tasks/metadata/additionalDetails.php]: Yes, thanks for notice me. The PR is not complete yet

**Comment by samus-aran** [comment]
Still Issues:
(would recommend installing the SonarLint on your IDE to review if changes have resolved the criticals (ask Daniel).

ref: https://sonarqube.com/issues#issues=AVajt02EcLjTdyDsXJWL
ref: https://sonarqube.com/issues#issues=AVajt02EcLjTdyDsXJWM

(I've gotten up to https://github.com/salesagility/SuiteCRM/pull/3207/commits/2982e2fd7c4dea65bd6e8519c614be8c3fd9c8f7)

File: data/SugarBean.php [hunk 2]
```diff
@@ -611,11 +611,11 @@ public static function createRelationshipMeta($key, $db, $tablename, $dictionary
...
      * Internal Function, do not override.
      */
-    public static function get_union_related_list($parentbean = null, $order_by = "", $sort_order = '', $where = "",
-                                                  $row_offset = 0, $limit = -1, $max = -1, $show_deleted = 0, $subpanel_def)
+    public static function get_union_related_list($parentbean, $order_by = "", $sort_order = '', $where = "",
+                                                  $row_offset = 0, $limit = -1, $max = -1, $show_deleted = 0, $subpanel_def = null)
     {
-        if (empty($parentbean)) {
-            $GLOBALS['log']->fatal('empty parent bean');
+        if (is_null($subpanel_def)) {
```
Comment (samus-aran) [inline data/SugarBean.php]: Don't think this is correct.... will subpanel_def be null?  Can't we both have parentbean = null and subpanel_def = null?

File: data/SugarBean.php [hunk 3]
```diff
@@ -906,11 +906,11 @@ protected static function build_sub_queries_for_union($subpanel_list, $subpanel_
      * @param array $secondary_queries
      * @return array $fetched data.
      */
-    public function process_union_list_query($parent_bean = null, $query = '',
-                                             $row_offset = 0, $limit = -1, $max_per_page = -1, $where = '', $subpanel_def, $query_row_count = '', $secondary_queries = array())
+    public function process_union_list_query($parent_bean, $query,
+                                             $row_offset, $limit = -1, $max_per_page = -1, $where = '', $subpanel_def = null, $query_row_count = '', $secondary_queries = array())
```
Comment (samus-aran) [inline data/SugarBean.php]: Same for the above comment.

**Comment by samus-aran** [comment]
All other commits looks resolved.

**Pull request merged**